ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -147,7 +161,14 @@ def observation_to_np_array(
         img = np.reshape(img, obs.shape)
         return img
     else:
-        img = process_pixels(obs.compressed_data, expected_channels)
+        if "compressed_channel_mapping" in (f.name for f in obs.DESCRIPTOR.fields):
+            img = process_pixels(
+                obs.compressed_data,
+                expected_channels,
+                list(obs.compressed_channel_mapping),
+            )
+        else:
+            img = process_pixels(obs.compressed_data, expected_channels)
```

Comment:
I think you can do

```
compressed_channel_mapping = list(obs.compressed_channel_mapping) if "compressed_channel_mapping" in (f.name for f in obs.DESCRIPTOR.fields) else None
img = process_pixels(
                obs.compressed_data,
                expected_channels,
                lcompressed_channel_mapping,
            )
```
If it makes it clearer.

Reply:
I'm not sure that checking for it in the fields is safe. What's the problem with just calling 
```
img = process_pixels(
  obs.compressed_data,
  expected_channels,
  obs.compressed_channel_mapping,
)
```
? If it's just the mypy type, there should be an interface that matches (maybe `Sequence`?).

Reply:
Wasn't sure if it's safe to access a unset field but it seems fine.